src/targets/gpu/loop.cpp

```diff
@@ -0,0 +1,134 @@
+#include <iterator>
+#include <migraphx/gpu/loop.hpp>
+#include <migraphx/gpu/context.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+shape hip_loop::compute_shape(std::vector<shape> inputs, std::vector<module_ref> mods) const
+{
+    inputs.pop_back();
+    inputs.pop_back();
+    inputs.erase(inputs.begin() + 3);
+    inputs.erase(inputs.begin() + 1);
+    return op.compute_shape(inputs, mods);
+}
+
+static std::pair<int, bool> get_name_index(const std::string& name, const std::string& param_prefix)
+{
+    auto loc = name.find(param_prefix);
+    if(loc != std::string::npos)
+    {
+        int index = std::stoi(name.substr(loc + param_prefix.size()));
+        return {index, true};
+    }
+
+    std::string out_prefix = "#output_";
+    loc                    = name.find(out_prefix);
+    if(loc != std::string::npos)
+    {
+        int index = std::stoi(name.substr(loc + out_prefix.size()));
+        return {index, false};
+    }
+
+    return {-1, false};
+}
+
+argument
+hip_loop::compute(const shape&,
+                  const std::vector<argument>& args,
+                  const std::vector<module_ref>& mods,
+                  const std::function<std::vector<argument>(
+                      module_ref&, const std::unordered_map<std::string, argument>&)>& run) const
+{
+    auto cpy_args = args;
+    std::vector<argument> cpu_args;
+    cpu_args.push_back(cpy_args.at(1));
+    cpu_args.push_back(cpy_args.at(3));
```

Comment:
I dont see the point of this vector. I think it would be better to have variable names for each cpu_arg:

```cpp
const auto& cond_arg = args.at(1));
```

Reply:
fixed